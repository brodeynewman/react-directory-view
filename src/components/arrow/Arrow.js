import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import styles from './arrow.css';

const Arrow = ({
  onExpand, onContract, isCollapsed, arrowStyles,
}) => (
  <span
    className={styles.arrow}
    role="presentation"
    onClick={!isCollapsed ? onExpand : onContract}
    style={{
      ...arrowStyles,
      transform: `rotate(${isCollapsed ? '135deg' : '45deg'})`,
    }}
  />
);

Arrow.propTypes = {
  arrowStyles: PropTypes.objectOf(PropTypes.any),
  isCollapsed: PropTypes.bool,
  onExpand: PropTypes.func,
  onContract: PropTypes.func,
};

Arrow.defaultProps = {
  arrowStyles: {},
  isCollapsed: false,
  onExpand: _.noop,
  onContract: _.noop,
};

export default Arrow;

