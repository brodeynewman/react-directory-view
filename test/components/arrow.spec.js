import React from 'react';
import renderer from 'react-test-renderer';
import Arrow from '../../src/components/arrow/Arrow';

describe('Arrow', () => {
  it('Arrow renders correctly', () => {
    const tree = renderer
      .create(<Arrow />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
