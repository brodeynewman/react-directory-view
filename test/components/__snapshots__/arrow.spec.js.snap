// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow Arrow renders correctly 1`] = `
<span
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "border": "4px solid transparent",
      "borderRight": "4px solid",
      "borderTop": "4px solid",
      "boxSizing": "border-box",
      "color": "#666",
      "cursor": "pointer",
      "display": "inline-block",
      "height": 0,
      "margin": 15,
      "position": "relative",
      "transform": "rotate(45deg)",
      "verticalAlign": "middle",
      "width": 0,
    }
  }
/>
`;
